### ISSUE TITLE
Service shuts down on startup with 504 Gateway Timeout

### ISSUE THREAD

**Issue by CamSoper** [wez/govee2mqtt#254]
### Govee Device SKU

any

### Govee2MQTT Version

2024.07.21-c9d27764

### Describe the issue

govee2mqtt is currently failing on startup with a 504 Gateway Timeout from the API.

Based on users currently reporting Govee app issues on [this old resurrected Reddit thread](https://www.reddit.com/r/Govee/comments/17kxeyw/keeps_saying_network_issue_but_everything_else/), it appears to be a server-side issue and no remediation in govee2mqtt should be required. I'm opening an issue for tracking and to provide a searchable artifact if it happens in the future. I'll close the issue when the issue clears up.

### Startup Diagnostics

```
govee2mqtt | [2024-07-30T19:31:50 INFO  govee::commands::serve] Starting service. version 2024.07.21-c9d27764
govee2mqtt | [2024-07-30T19:31:50 INFO  govee::commands::serve] Querying platform API for device list
govee2mqtt | Error: request https://openapi.api.govee.com/router/api/v1/user/devices status 504: Gateway Timeout. Response body: <html>
govee2mqtt | <head><title>504 Gateway Time-out</title></head>
govee2mqtt | <body>
govee2mqtt | <center><h1>504 Gateway Time-out</h1></center>
govee2mqtt | </body>
govee2mqtt | </html>
govee2mqtt |
govee2mqtt |
govee2mqtt | Stack backtrace:
govee2mqtt |    0: anyhow::error::<impl anyhow::Error>::msg
govee2mqtt |    1: anyhow::__private::format_err
govee2mqtt |    2: govee::cache::CacheResult<T>::into_result
govee2mqtt |    3: govee::platform_api::GoveeApiClient::get_devices::{{closure}}
govee2mqtt |    4: govee::commands::serve::ServeCommand::run::{{closure}}
govee2mqtt |    5: govee::Args::run::{{closure}}
govee2mqtt |    6: tokio::runtime::park::CachedParkThread::block_on
govee2mqtt |    7: tokio::runtime::context::runtime::enter_runtime
govee2mqtt |    8: tokio::runtime::runtime::Runtime::block_on
govee2mqtt |    9: govee::main
govee2mqtt |   10: std::sys_common::backtrace::__rust_begin_short_backtrace
govee2mqtt |   11: std::rt::lang_start::{{closure}}
govee2mqtt |   12: std::rt::lang_start_internal
govee2mqtt |   13: main
```

**Comment by dog5point0** [comment]
Having the same issue here.

**Comment by Rodney-Smith** [comment]
Also having the issue with openapi.api.govee.com
curl --request GET \
     --url https://openapi.api.govee.com/v1/devices \
         --header 'Govee-API-Key: <your-key-here>' \
     --header 'accept: application/json'

<html>
<head><title>504 Gateway Time-out</title></head>
<body>
<center><h1>504 Gateway Time-out</h1></center>
</body>
</html>

This URL of developer-api.govee.com seems to be functioning just fine
curl --request GET \
     --url https://developer-api.govee.com/v1/devices \
     --header 'Govee-API-Key: <your-key-here>' \
     --header 'accept: application/json'

{"data":{"devices":[]...,"message":"Success","code":200}

**Comment by Nitnomart** [comment]
I've noticed the same issue, clearly a Govee side issue, hopefully just temporary and not a breaking change for the integration.

With local LAN control enabled on the devices and govee2mqtt talking locally to the devices is there a way to avoid issues should the internet go down or some other Govee issue like this? Is this integration 100% reliant on the cloud hosted Govee API to function? Or was I unlucky because I restarted HA while Govee had an issue and then purged caches in an attempt to get this working?

Is govee2mqtt simply trying to speak to the Govee cloud API for a list of devices to then talk locally to? 

For info there is a similar more recent Redit post here talking about server issues with many people reporting them with the standard Govee app: https://www.reddit.com/r/Govee/comments/1eg6rj1/network_error/

Interestingly my Govee app works fine but I believe I am able to control my devices via Bluetooth at the moment rather than Wi-Fi if I'm correct.
